modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/CastResolutionTest.java

```diff
@@ -50,6 +51,14 @@ public class CastResolutionTest extends AbstractPlannerTest {
 
     private static final Set<String> dtNames = DATETIME_TYPES.stream().map(SqlTypeName::getName).collect(Collectors.toSet());
 
+    private static final Set<String> ymInterval = Set.of(INTERVAL_YEAR.getName(), INTERVAL_MONTH.getName());
+
+    private static final Set<String> dayInterval = Set.of(INTERVAL_HOUR.getName(), INTERVAL_MINUTE.getName());
```

Comment:
Why `INTERVAL_DAY`/`INTERVAL_SECOND` intervals are omitted?

Reply:
good catch ))) i just don`t want to append intervals like INTERVAL_DAY_MINUTE, INTERVAL_MINUTE_SECOND for more test simplicity. SqlTypeName class contains collection wrappers: YEAR_INTERVAL_TYPES, DAY_INTERVAL_TYPES and i decide that to check one type from this wrappers would be enough.

Reply:
additional tests appended